server/sonar-web/src/main/js/apps/component-measures/details/history/MeasureHistory.js

```diff
@@ -110,9 +110,10 @@ export default class MeasureHistory extends React.PureComponent {
     }
 
     const data = snapshots.map(snapshot => {
+      const valueAsNumber = Number(snapshot.value);
       return {
         x: snapshot.date,
-        y: Number(snapshot.value)
+        y: isNaN(valueAsNumber) ? snapshot.value : valueAsNumber
```

Comment:
This could potentially hide an error, when a number is badly converted for some reason. Instead I'd make this condition more clear by checking the metric type, something like 
```javascript
this.props.metric.type === 'LEVEL' ? snapshot.value : Number(snapshot.value)
```